e2e/cluster.go

```diff
@@ -6,30 +6,64 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
 )
 
 func setupCluster(ctx context.Context, cloud *azureClient, location, resourceGroupName, clusterName string) error {
-	aksCluster, err := cloud.aksClient.Get(ctx, resourceGroupName, clusterName, nil)
-	if err != nil {
-		return fmt.Errorf("failed to get aks cluster: %q", err)
-	}
+	var needNewCluster bool
 
-	rgExistence, err := cloud.resourceGroupClient.CheckExistence(ctx, agentbakerTestResourceGroupName, nil)
+	rgExistence, err := cloud.resourceGroupClient.CheckExistence(ctx, resourceGroupName, nil)
 	if err != nil {
-		return fmt.Errorf("failed to get MC RG: %q", err)
+		return fmt.Errorf("failed to get AB E2E RG %q: %q", resourceGroupName, err)
 	}
 
```

Comment:
Just to confirm, is it possible for err==nil and rgExistence.Success to be false? Based on error message above it seems like non-existent RG will lead to an error, and exit the function before?

Reply:
this function specifically checks for existence of the resource group, rather than just a plain get - my understanding is that if the RG doesn't exist then err will indeed be nil, and will be non-nil due to any other sort of error. I tested this and it seems to be working fine.